docs/rfc/0698-cli-infrastructure.adoc

```diff
@@ -0,0 +1,297 @@
+binary MUST follow the naming convention of `rad-*`. To reiterate
+here, these new commands MUST NOT override the core set outlined here
+(nor in further RFCs).
+
+== Global Parameters
+
+Many subcommands will likely want to share a set of options. These
+options, dubbed as "global parameters" here, will be a defined set --
+exported from `radicle-link` for re-use -- that the base `rad` command
+will always receive and pass onto the subcommand being called.
+
+Other subcommands that are written in the target language (currently
+Rust) will be able to import the `struct` for reuse. If the subcommand
+is in another target language it SHALL add these parameters to its
+parameter list prefixed with `global-`. The `rad` command will parse
+these parameters, setting defaults if necessary, and SHALL pass them
+varbatim to the subcommand.
+
+Thus, all subcommands SHALL receive these global parameters as part of
+their interface. We list these parameters here, but it is NOT an
+exhaustive list:
+
+* `global-verbose`
+* `global-quiet`
+* `global-profile := active (default) | <profile-id>`
+
+== CLI Events
+
+The execution of a CLI subcommand could potentially result in an
+interesting event. For example, if Upstream is
+running and I create a project via the CLI, then a good user
+experience is that the Upstream is updated with the new project
+(without refreshing). In this case, the event is the creation of the
+project.
+
+In <<arch-pubsub, RFC-682#PubSub>>, a PubSub component was proposed --
+with a detailed specification being left to a future RFC. We will
+assume the existence of this component here. The CLI commands in this
+RFC MAY publish interesting events to the PubSub component. This will
+allow other applications to react upon seeing these events via
+subscriptions to the PubSub component.
+
+Other subcommands MAY also publish interesting events, as long as they
+adhere to the specification of the future RFC. They SHALL reuse any
+events defined as part of the <<Core>> commands. That is to say, there
+will be a core set of events that should not be overridden, but rather
+reused if needs be.
+
+== Key Access
+
+<<arch-sigs, RFC-682#Signatures>> outlined the signing of Radicle data. This is essential to
+this RFC as commands that write to the storage generally need access
+to the secret key, whether for signing data or proof of ownership to
+the storage. For most (if not all) read-only commands, no access to
+the secret key is needed.
+
+To achieve the goals of having a read-only storage and using
+`<<ssh-agent>>` for signing, we propose the following should be
+implemented:
+
+. `ssh-agent` compatability with `<<radicle-keystore>>`, which has been
+started in `<<rk-17, radicle-keystore#17>>`.
+. A subcommand `rad-profile-ssh-add` for adding one's secret key to the
+`ssh-agent` using the supplied constraints, e.g. ask for password.
+. Implement `<<rl-461, radicle-link#461>>` to have a read-only storage.
```

Comment:
This issue is closed :thinking: https://github.com/radicle-dev/radicle-link/issues/461

Reply:
It wasn't at the time of writing :) I'll remove it now.